lib/airtable/entity/record.rb

```diff
@@ -0,0 +1,94 @@
+require 'time'
+# Main object for store Airtable Record entity
+module Airtable
+  module Entity
+    # Airtable Record entity
+    class Record
+      extend Forwardable
+      attr_reader :id, :created_at, :fields
+
+      def_delegators :@fields, :[], :[]=
+
+      def initialize(id, options = {})
+        @id = id
+        parse_options(options)
+      end
+
+      def new_record?
+        @id.nil? || @id.empty?
+      end
+
+      def __create__(base, name)
+        res = base.__make_request__(:post, name, fields: fields)
+        @id = res['id']
+        parse_options(fields: res['fields'], created_at: res['createdTime'])
+        self
+      end
+
+      def __update__(base, name)
+        args = [:patch, [name, @id].join('/'), fields: fields]
+        res  = base.__make_request__(*args)
+        parse_options(fields: res['fields'])
+        self
+      end
+
+      def __fetch__(base, path)
+        res = base.__make_request__(:get, path, {})
+        parse_options(fields: res['fields'], created_at: res['createdTime'])
+        self
+      end
+
+      def __replace__(base, name)
+        res = base.__make_request__(:put, [name, @id].join('/'), fields: fields)
+        parse_options(fields: res['fields'])
+        self
+      end
+
+      def __destroy__(base, name)
+        res = base.__make_request__(:delete, [name, @id].join('/'), {})
+        res['deleted']
+      end
+
+      def [](key)
+        @fields[key.to_s]
+      end
+
+      def []=(key, value)
+        @fields[key.to_s] = value
+      end
+
+      class << self
+        def all(base, name, params)
```

Comment:
It seems that `all` and `__fetch__` should be class methods on Table, not Record.

Reply:
No. `Table` class is already too big. And better to use such methods in `Record` to follow the Ruby/Ruby on Rails way.

Reply:
sounds good